---
title: "DESeq2_CMC_Tutorial_Part4"
author: "Nickie Safarian"
date: "11/23/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

Here, we will show how to assess the correlation between the expression of select.genes and SST cell type proportion as well as SST.PRS.Scores (generated by Alex). 


## Load the packages

```{r, include=FALSE}

library(tidyverse)
library(limma)
library(Glimma)
library(edgeR)


```


## Prepare the Metadata

```{r , echo=FALSE, include=FALSE}

# get CMC clinical meta
cmc_clinical_path = '/external/rprshnas01/netdata_kcni/stlab/Public/commonmind/ControlledAccess/Data/Clinical/'
cmc_clinical_csv = paste0(cmc_clinical_path, 'CMC_Human_clinical_metadata.csv')


# get CMC rnaseq meta
cmc_rnaseq_path = '/external/rprshnas01/netdata_kcni/stlab/Public/commonmind/ControlledAccess/Data/RNAseq/Release4/Metadata/'
cmc_rnaseq_meta_path = paste0(cmc_rnaseq_path, 'CMC_Human_rnaSeq_metadata.csv')

cmc_rnaseq_meta = read_csv(cmc_rnaseq_meta_path)
colnames(cmc_rnaseq_meta) = make.names(colnames(cmc_rnaseq_meta), unique = T)

cmc_clinical_meta = read_csv(cmc_clinical_csv)
colnames(cmc_clinical_meta) = make.names(colnames(cmc_clinical_meta), unique = T)


# create a new df called cmc_meta that merges the clinical and rnaseq meta - note that this duplicates data from some subjects
cmc_meta = full_join(cmc_clinical_meta, cmc_rnaseq_meta)

# factor the variables of interest
cmc_meta$Dx = factor(cmc_meta$Dx, levels = c('Control', 'SCZ', 'BP', 'AFF', 'undetermined'))
cmc_meta$Reported.Gender = factor(cmc_meta$Reported.Gender, levels = c('Male', 'Female'))
cmc_meta$Ethnicity = factor(cmc_meta$Ethnicity, levels = c('Caucasian', 'African-American', 'Asian', 'Hispanic', '(Multiracial)'))


# convert ages to numbers
cmc_meta = cmc_meta %>% mutate(Age_norm = case_when(Age.of.Death == '90+' ~ 90, 
                                                    TRUE ~ as.numeric(Age.of.Death)))

# get numeric subject id - this is required for mapping to etienne's data from Pitt cohort
cmc_meta = cmc_meta %>% mutate(Subject = parse_number(SampleID))

```


## Import Alex's PRS file

```{r, include=FALSE}

# load in Alex's estimates of PRSs for SZ, BP, etc. (different Diagnosis)
datit=read.csv('/external/rprshnas01/netdata_kcni/stlab/CMC_genotypes/SNPs/Release3/Metadata/CMC_Human_SNP_metadata.csv')

datit <- datit %>% dplyr:: select (., Individual_ID, Genotyping_Sample_ID)

prs=read_tsv("/external/rprshnas01/netdata_kcni/stlab/Alex_PRS/CMC_genotypes/PRSCS_cmcrodrigo")

prs$ID=gsub("0_", "", prs$ID)

datmrg1=merge(datit,prs,by.x = 'Genotyping_Sample_ID', by.y='ID')
# datcmc1=merge(datcmc,datmrg1,by.x="Individual.ID",by.y="Individual_ID")


# merge the metdata, PRS, and the Genotype data
joined_df = left_join(cmc_meta, datmrg1, by = c("Individual.ID" ="Individual_ID")) 

# Plot it 
joined_df %>% ggplot(aes(x = Dx, y = SZ)) + geom_boxplot() + facet_wrap(~Institution)

```

# SST PRSs - based on Alex's estimates from CMC based on Fernanda's code

```{r, include=FALSE}

sst_prs = read_delim('/external/rprshnas01/kcni/asegura/SST_Fernanda/results/CMC_PRS_SST.txt', delim = '\t')
  
sst_prs$ID = str_sub(sst_prs$ID, 3)


# Add SST_PRS data to the previous joined_df
joined_df = left_join(joined_df, sst_prs, by = c("Genotyping_Sample_ID" = "ID"))

```


## Now load in our data from CMC RNAseq data

```{r, include=FALSE}

### Now read in gene expression data from CommonMind subjects
gene_counts_ob = readRDS('/external/rprshnas01/netdata_kcni/stlab/Public/commonmind/ControlledAccess/Data/RNAseq/Release4/QuantitatedExpression/geneCountsMerged.RDS')

# this is the expression data from Pitt Penn and MSSM
cmc_expr_1 = read_tsv('/external/rprshnas01/netdata_kcni/stlab/Public/commonmind/ControlledAccess/Data/RNAseq/Release4/QuantitatedExpression/MSSM.Penn.Pitt_DLPFC.featureCount.tsv.gz')
# 
cmc_expr_2 = read_tsv('/external/rprshnas01/netdata_kcni/stlab/Public/commonmind/ControlledAccess/Data/RNAseq/Release4/QuantitatedExpression/NIMH.HBCC.featureCount.tsv.gz')
# 
cmc_expr = cbind(cmc_expr_1, cmc_expr_2[, 7:387]) 

cmc_expr_counts_only = cmc_expr[, 7:1011] %>% as.data.frame()
rownames(cmc_expr_counts_only) = cmc_expr$Geneid %>% substr(., 1, 15) %>% make.names(., unique = T)

```


### perform normalization for gene expression data from CMC

```{r}

cmc_expr_cpm = edgeR::cpm(cmc_expr_counts_only, prior.count = 1)
cmc_expr_cpm = cmc_expr_cpm %>% as.data.frame() %>% tibble::rownames_to_column(var = "ensembl_id")

sst_mrna = cmc_expr_cpm %>% filter(ensembl_id == 'ENSG00000157005') %>% t() %>% as.data.frame() %>%  tibble::rownames_to_column(var = 'SampleID') 

sst_mrna = sst_mrna[-1, ]
   
colnames(sst_mrna) = c('SampleID', 'SST')


# This will join metdata, PRS(for diagnosis), Genotype, and SST_mrna datasets:
joined_df = left_join(joined_df, sst_mrna)

```

## Plot the SST gene expression vs. Diagnosis

```{r, include=FALSE}

str(joined_df)
# SST gene counts is stored as character , let's convert it to numeric for plotting
joined_df$SST <- as.numeric(joined_df$SST) 


joined_df %>% group_by(Dx) %>% 
  ggplot(aes(x = Dx, y = SST)) + 
  geom_boxplot()+
  geom_jitter(shape=1, color="skyblue2", position=position_jitter(0.2))+
  labs(x= "Diagnosis", y="SST_gene_counts")+
  theme(axis.text = element_text(size=10, face="bold"),
        axis.title = element_text(size=10, face="bold"))+
  theme_bw()

```


## Design a linear model and explore the correlation between SST gene counts and other covariates

```{r}

library(broom)
# First, scale the numerical covariates
joined_df$PMI..in.hours. <- scale(joined_df$PMI..in.hours.)
joined_df$SST_PRS_0.1 <- scale(joined_df$SST_PRS_0.1)
joined_df$Age_norm <- scale(joined_df$Age_norm)

model_effects = joined_df %>% filter(!Institution == 'NIMH-HBCC', 
                     Dx %in% c('Control', 'SCZ','BP'), 
                     Sex %in% c('XY', 'XX')) %>% 
  # group aging_paper_label stacked data by cell_type
  group_by(Institution) %>%
  
  # fit all the cell_type_prop data according to the model 
  # using the broom package to tidy the results 
  do(tidy(lm(SST ~ Age_norm  +  
               PMI..in.hours. + 
               Sex + Dx + SST_PRS_0.1,  data = .))) %>%
  
  # unstack the data and adjust for multiple comparisons using the Benjamini-Hochberg method
  ungroup() %>% 
  mutate(padj = p.adjust(`p.value`, method = 'BH')) %>%
  
  # clean up the names the the term column
  mutate(term = recode(term,
                       `(Intercept)` = "Intercept",
                       `SexM` = "gender:Male",
                       `DxSCZ` = "SCZ",
                       `DxBP` = "BP",
                       `DxMDD` = "MDD",
                       `EthnicityAfrican-American` = "ethnicity:Black", ## Q in the model ???
                       `scale(Age_norm)` = "age",
                       `scale(PMI..in.hours.)` = "PMI", 
                       `scale(SST_PRS_0.1)` = 'SST_PRS_0.1', 
                       `scale(SST_PRS_1)` = 'SST_PRS_1'))
```


## Visualize

```{r}

model_effects %>% filter(term == 'SST_PRS_0.1') %>% 
  ggplot(aes(x = Institution, y = estimate)) + 
  geom_bar(stat = 'identity', width = 0.8, aes(y = estimate), colour = "black") +
  geom_errorbar((aes(ymin = estimate - std.error, ymax = estimate + std.error)))

```



## Extra codes for modeling the data

```{r}


model1 <- joined_df %>% filter(!Institution == 'NIMH-HBCC', 
                     Dx %in% c('Control', 'SCZ','BP'), 
                     Sex %in% c('XY', 'XX')) %>% 
  lm(SST ~ SST_PRS_0.1 + PMI..in.hours. + Age_norm + Sex + Dx, data=.)

anova(model1)
summary(model1)
  
plot(model1, las = 1) 
par(mfrow = c(2,2), oma = c(0, 0, 1.1, 0))

```



```{r}

## Scatterplot of SST expression vs PRS (shown for p-value threshold of 0.5 from Example (a))

plot(joined_df$SST_PRS_0.1, joined_df$SST,  ylab="SST gene expression", xlab= "SST_PRS 0.1")
abline(lm(joined_df$SST ~ joined_df$SST_PRS_0.1), col = "red",
lty = 1, lwd = 1)

```


```{r}

########################################
model2 = joined_df %>% filter(!Institution == 'NIMH-HBCC', 
                     Dx %in% c('Control', 'SCZ','BP'), 
                     Sex %in% c('XY', 'XX')) %>%
  group_by(Dx) %>%
  do(tidy(lm(SST ~ Age_norm  +  
               PMI..in.hours. + 
               Sex + SST_PRS_0.1,  data = .))) %>%
  ungroup() %>% 
  mutate(padj = p.adjust(`p.value`, method = 'BH'))


model2 %>% filter(term == 'SST_PRS_0.1') %>% 
  ggplot(aes(x = Dx, y = estimate)) + 
  geom_bar(stat = 'identity', width = 0.8, aes(y = estimate), fill = "grey55") +
  geom_errorbar((aes(ymin = estimate - std.error, ymax = estimate + std.error)))


```








